### ISSUE TITLE
Cinnamon Settings Daemon crashes on login with NVIDIA drivers [Workaround]

### ISSUE THREAD

**Issue by emiliedancer** [linuxmint/cinnamon-settings-daemon#52]
Running:
  - Linux Mint 17 Qiana 
  - Cinnamon 2.2.13
  - NVIDIA 337.25
  - MDM 1.6.9

Has been running fine since Qiana came out (almost 2 months).  For the first time, I plugged in my tv via HDMI cable, everything worked fine, I changed to the appropriate resolution using Nvidia X Server Settings and all worked as expected.

However, after unplugging tv and returning to normal resolution, cinnamon-settings-daemon now crashes on login.  Symptoms:
 - Boot up to MDM works fine
 - After entering login credentials, screen goes black for a long time (10-14 seconds)
 - Cinnamon loads with hardware rendering but with very ugly default theme (icons, controls, windows, etc.).  The terminal is black background with black text so impossible to use. 
 - Trying to change things in theme settings has no effect
 - Happens every login

Looking at syslog reveals that cinnamon-setttings-daemon is crashing on startup ( notice 14 second delay in login) and causing a kernel trap (?) and a dead whale.  (I deleted some lines where colord and icc profiles were loaded because there were no errors):

From syslog:
    Jul 27 21:52:17 Tehanu kernel: [   29.442413] traps: cinnamon-settin[2252] trap int3 ip:7f1f8099bc13 sp:7fff71050b50 error:0
    Jul 27 21:52:17 Tehanu cinnamon-session[1993]: WARNING: Application 'cinnamon-settings-daemon.desktop' killed by signal 5
    Jul 27 21:52:24 Tehanu NetworkManager[903]: <warn> nl_recvmsgs() error: (-33) Dump inconsistency detected, interrupted
    Jul 27 21:52:31 Tehanu cinnamon-session[1993]: WARNING: Application 'cinnamon-settings-daemon.desktop' failed to register before timeout
    Jul 27 21:52:31 Tehanu cinnamon-session[1993]: CRITICAL: We failed, but the fail whale is dead. Sorry....

I would blame NVIDIA but it has been working fine for almost 2 months.  I guess it is an interaction between the Cinnamon and the NVIDIA driver triggered by changing the resolution to something new and back? 

Below is inxi information in case it is helpful.

$  cinnamon --version; inxi -Fxz
    Cinnamon 2.2.13
    System:    Host: Tehanu Kernel: 3.13.0-24-generic x86_64 (64 bit, gcc: 4.8.2) Desktop: Gnome Distro: Linux Mint 17 Qiana
    Machine:   System: Micro-Star product: GS70 2OD version: REV:1.0
               Mobo: Micro-Star model: MS-1771 version: REV:0.B Bios: American Megatrends version: E1771IMS.70C date: 10/01/2013
    CPU:       Quad core Intel Core i7-4700HQ CPU (-HT-MCP-) cache: 6144 KB flags: (lm nx sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx) bmips: 19157.1
               Clock Speeds: 1: 800.00 MHz 2: 800.00 MHz 3: 800.00 MHz 4: 800.00 MHz 5: 1600.00 MHz 6: 800.00 MHz 7: 800.00 MHz 8: 800.00 MHz
    Graphics:  Card: Intel 4th Gen Core Processor Integrated Graphics Controller bus-ID: 00:02.0
               X.Org: 1.15.1 driver: nvidia Resolution: 1920x1080@60.0hz
               GLX Renderer: GeForce GTX 765M/PCIe/SSE2 GLX Version: 4.4.0 NVIDIA 337.25 Direct Rendering: Yes
    Audio:     Card: Intel 8 Series/C220 Series Chipset High Definition Audio Controller driver: snd_hda_intel bus-ID: 00:1b.0
               Sound: Advanced Linux Sound Architecture ver: k3.13.0-24-generic
    Network:   Card-1: Qualcomm Atheros AR9462 Wireless Network Adapter driver: ath9k bus-ID: 05:00.0
               IF: wlan0 state: up mac: <filter>
               Card-2: Qualcomm Atheros Killer E2200 Gigabit Ethernet Controller driver: alx port: c000 bus-ID: 04:00.0
               IF: eth0 state: down mac: <filter>
               Card-3: Atheros driver: usb usb-ID: 0cf3:3004
               IF: N/A state: N/A speed: N/A duplex: N/A mac: N/A
    Drives:    HDD Total Size: 1640.4GB (58.3% used) 1: id: /dev/sda model: TOSHIBA_THNSNF12 size: 128.0GB
               2: id: /dev/sdb model: SAMSUNG_MZMTD512 size: 512.1GB 3: id: /dev/sdc model: HGST_HTS721010A9 size: 1000.2GB
    Partition: ID: / size: 455G used: 424G (99%) fs: ext4 ID: swap-1 size: 16.39GB used: 0.00GB (0%) fs: swap
    RAID:      No RAID devices detected - /proc/mdstat and md_mod kernel raid module present
    Sensors:   System Temperatures: cpu: 29.8C mobo: 27.8C gpu: 0.0:52C
               Fan Speeds (in rpm): cpu: N/A
    Info:      Processes: 205 Uptime: 2 min Memory: 1006.7/15967.7MB Runlevel: 2 Gcc sys: 4.8.2 Client: Shell inxi: 1.8.4

**Comment by fcapano** [comment]
I can confirm that this problem still exists on Cinnamon 2.4.6 (Linux Mint 17.1 Rebecca)

**Comment by mahadi** [comment]
Same for me: I can confirm that this problem still exists on Cinnamon 2.4.6 (Linux Mint 17.1 Rebecca)

My monitors.xml:

<monitors version="1">
  <configuration>
      <clone>no</clone>
      <output name="LVDS-1-0">
          <vendor>CMN</vendor>
          <product>0x1720</product>
          <serial>0x00000000</serial>
      </output>
      <output name="VGA-1-0">
      </output>
      <output name="HDMI-1-0">
          <vendor>GSM</vendor>
          <product>0x587e</product>
          <serial>0x01010101</serial>
          <width>1920</width>
          <height>1080</height>
          <rate>0</rate>
          <x>0</x>
          <y>0</y>
          <rotation>normal</rotation>
          <reflect_x>no</reflect_x>
          <reflect_y>no</reflect_y>
          <primary>yes</primary>
      </output>
      <output name="DisplayPort-1-0">
      </output>
  </configuration>
</monitors>

**Comment by pickardjoe** [comment]
Same for me: I can confirm that this problem still exists on Cinnamon 2.4.6 (Ubuntu 14.04 Trusty)

**Comment by pickardjoe** [comment]
I've found a workaround.

1. Open 'dconf-editor'.
2. Navigate to 'org.cinnamon.settings-daemon.plugins.xrandr'
3. Uncheck the "active" box.
4. Configure your resolutions with the 'nvidia-settings' tool.
5. Reboot
6. Some text items such as the "Menu" button text might be in an unusually large font. To fix them, keep reading (and doing).
7. Right click the bottom panel.
8. Select "Panel Settings".
9. Ensure the following boxes are checked:
![screenshot from 2015-02-05 06 55 33](https://cloud.githubusercontent.com/assets/1213486/6059843/3d9eaee2-ad04-11e4-9ef5-b021284b67f0.png)
10. Repeat steps 7 - 9 for any other distorted panels.

**Comment by PragTob** [comment]
Broken over here as well, only after trying to use the recommended nvidia driver (I have a GT 750) - the workaround above here worked but seemingly xrandr wasn't active anymore and that seemed to have interfered with me changing resolutions which I need to do frequently (Laptop, working at different client sites). I reverted back to nouveau now.

Some information:

```
tobi@airship ~ $ cinnamon --version; inxi -Fxz
Cinnamon 2.4.8
System:    Host: airship Kernel: 3.13.0-37-generic x86_64 (64 bit, gcc: 4.8.2) 
           Desktop: Gnome  (Gtk 2.24.23) Distro: Linux Mint 17.1 Rebecca
Machine:   System: Dell (portable) product: XPS 15 9530 version: A07
           Mobo: Dell model: XPS 15 9530 version: A07 Bios: Dell version: A07 date: 10/15/2014
CPU:       Quad core Intel Core i7-4712HQ CPU (-HT-MCP-) cache: 6144 KB flags: (lm nx sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx) bmips: 18357.2 
           Clock Speeds: 1: 800.00 MHz 2: 800.00 MHz 3: 800.00 MHz 4: 800.00 MHz 5: 800.00 MHz 6: 800.00 MHz 7: 1700.00 MHz 8: 800.00 MHz
Graphics:  Card: Intel 4th Gen Core Processor Integrated Graphics Controller bus-ID: 00:02.0 
           X.Org: 1.15.1 drivers: (unloaded: fbdev,vesa) FAILED: nouveau,intel Resolution: 1920x1080@59.9hz 
           GLX Renderer: Mesa DRI Intel Haswell Mobile GLX Version: 3.0 Mesa 10.1.3 Direct Rendering: Yes
Audio:     Card-1: Intel Xeon E3-1200 v3/4th Gen Core Processor HD Audio Controller driver: snd_hda_intel bus-ID: 00:03.0
           Card-2: Intel 8 Series/C220 Series Chipset High Definition Audio Controller driver: snd_hda_intel bus-ID: 00:1b.0
           Sound: Advanced Linux Sound Architecture ver: k3.13.0-37-generic
Network:   Card: Intel Wireless 7260 driver: iwlwifi ver: in-tree: bus-ID: 06:00.0
           IF: wlan0 state: up mac: <filter>
Drives:    HDD Total Size: 512.1GB (16.4% used) 1: id: /dev/sda model: LITEONIT_LMT size: 512.1GB 
Partition: ID: / size: 454G used: 78G (19%) fs: ext4 ID: /boot size: 236M used: 84M (38%) fs: ext2 
           ID: swap-1 size: 17.08GB used: 0.00GB (0%) fs: swap 
RAID:      No RAID devices detected - /proc/mdstat and md_mod kernel raid module present
Sensors:   System Temperatures: cpu: 27.8C mobo: N/A 
           Fan Speeds (in rpm): cpu: N/A 
Info:      Processes: 245 Uptime: 1 min Memory: 1307.0/15952.2MB Runlevel: 2 Gcc sys: 4.8.2 Client: Shell inxi: 1.8.4 
```

I'd be super glad to see this problem fixed... also experiencing the random freezes (appended nomodeset, hope that this will help).

**Comment by maxkl** [comment]
This issue affects me too. I am currently using cinnamon 2.4.8.
Here is the output of `inxi -Fxz`
```
System:    Host: fuchs Kernel: 3.13.0-37-generic x86_64 (64 bit, gcc: 4.8.2) Desktop: Gnome Distro: Linux Mint 17.1 Rebecca
Machine:   System: Hewlett-Packard (portable) product: HP Pavilion 15 Notebook PC version: 0896100000304100000620100
           Mobo: Hewlett-Packard model: 2166 version: 29.23 Bios: Insyde version: F.21 date: 08/08/2013
CPU:       Dual core Intel Core i5-4200U CPU (-HT-MCP-) cache: 3072 KB flags: (lm nx sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx) bmips: 9178.34 
           Clock Speeds: 1: 1600.00 MHz 2: 759.00 MHz 3: 759.00 MHz 4: 759.00 MHz
Graphics:  Card: Intel Haswell-ULT Integrated Graphics Controller bus-ID: 00:02.0 
           X.Org: 1.15.1 driver: nvidia Resolution: 1366x768@60.1hz 
           GLX Renderer: GeForce GT 740M/PCIe/SSE2 GLX Version: 4.4.0 NVIDIA 331.113 Direct Rendering: Yes
Audio:     Card-1: Intel Lynx Point-LP HD Audio Controller driver: snd_hda_intel bus-ID: 00:1b.0
           Card-2: Intel Haswell-ULT HD Audio Controller driver: snd_hda_intel bus-ID: 00:03.0
           Sound: Advanced Linux Sound Architecture ver: k3.13.0-37-generic
Network:   Card-1: Realtek RTL8101E/RTL8102E PCI Express Fast Ethernet controller 
           driver: r8169 ver: 2.3LK-NAPI port: 4000 bus-ID: 09:00.0
           IF: eth0 state: down mac: <filter>
           Card-2: Realtek RTL8188EE Wireless Network Adapter driver: rtl8188ee port: 5000 bus-ID: 08:00.0
           IF: wlan0 state: up mac: <filter>
Drives:    HDD Total Size: 500.1GB (6.7% used) 1: id: /dev/sda model: HGST_HTS545050A7 size: 500.1GB 
Partition: ID: / size: 20G used: 9.3G (50%) fs: ext4 ID: /home size: 99G used: 22G (24%) fs: ext4 
           ID: swap-1 size: 10.74GB used: 0.00GB (0%) fs: swap 
RAID:      No RAID devices detected - /proc/mdstat and md_mod kernel raid module present
Sensors:   System Temperatures: cpu: 51.0C mobo: N/A gpu: 0.0:51C 
           Fan Speeds (in rpm): cpu: N/A 
Info:      Processes: 190 Uptime: 13 min Memory: 1393.4/7757.3MB Runlevel: 2 Gcc sys: 4.8.2 Client: Shell inxi: 1.8.4
```

**Comment by zazasa** [comment]
Same issue for me on a fresh Mint 7.1 installation, I switched to intel as maxki did. 
Now i cannot switch to NVIDIA using the same interface:
![schermata del 2015-05-07 14 06 32-cut](https://cloud.githubusercontent.com/assets/6582479/7514930/9a4e00ce-f4c2-11e4-84ac-58cffe9f385e.png)

and inxi cannot show my NVIDIA card info, anyway it is a GT 750M:
```
Cinnamon 2.4.8
System:    Host: calvin Kernel: 3.16.0-37-generic x86_64 (64 bit, gcc: 4.8.2) Desktop: Gnome Distro: Linux Mint 17.1 Rebecca
Machine:   System: Alienware product: Alienware 14 version: A09
           Mobo: Alienware model: 07MJ2Y version: A00 Bios: Alienware version: A09 date: 04/23/2014
CPU:       Quad core Intel Core i7-4700MQ CPU (-HT-MCP-) cache: 6144 KB flags: (lm nx sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx) bmips: 19155.6 
           Clock Speeds: 1: 2549.625 MHz 2: 2388.375 MHz 3: 2400.00 MHz 4: 804.187 MHz 5: 2587.125 MHz 6: 2400.00 MHz 7: 2400.281 MHz 8: 2398.125 MHz
Graphics:  Card: Intel 4th Gen Core Processor Integrated Graphics Controller bus-ID: 00:02.0 
           X.Org: 1.15.1 drivers: intel (unloaded: fbdev,vesa) Resolution: 1920x1080@60.0hz, 1920x1080@60.0hz 
           GLX Renderer: Mesa DRI Intel Haswell Mobile GLX Version: 3.0 Mesa 10.1.3 Direct Rendering: Yes
Audio:     Card-1: Intel Xeon E3-1200 v3/4th Gen Core Processor HD Audio Controller driver: snd_hda_intel bus-ID: 00:03.0
           Card-2: Intel 8 Series/C220 Series Chipset High Definition Audio Controller driver: snd_hda_intel bus-ID: 00:1b.0
           Sound: Advanced Linux Sound Architecture ver: k3.16.0-37-generic
Network:   Card-1: Qualcomm Atheros AR9462 Wireless Network Adapter driver: ath9k bus-ID: 04:00.0
           IF: wlan0 state: up mac: <filter>
           Card-2: Qualcomm Atheros Killer E2200 Gigabit Ethernet Controller driver: alx port: 3000 bus-ID: 02:00.0
           IF: eth0 state: down mac: <filter>
Drives:    HDD Total Size: 256.1GB (2.6% used) 1: id: /dev/sda model: Samsung_SSD_840 size: 256.1GB temp: 0C 
Partition: ID: / size: 131G used: 6.3G (6%) fs: ext4 
RAID:      No RAID devices detected - /proc/mdstat and md_mod kernel raid module present
Sensors:   System Temperatures: cpu: 61.0C mobo: N/A 
           Fan Speeds (in rpm): cpu: N/A 
Info:      Processes: 229 Uptime: 34 min Memory: 1733.6/7870.0MB Runlevel: 2 Gcc sys: 4.8.2 Client: Shell inxi: 1.8.4 

```

**Comment by jebeaudet** [comment]
I had the same problem and updating the drivers to the latest from Nvidia site (346.59) seems to have solved the problem. Worth a try IMO.
```
System:    Host: DESKTOP-Linux Kernel: 3.16.0-37-generic x86_64 (64 bit, gcc: 4.8.2) 
           Desktop: Gnome Distro: Linux Mint 17.1 Rebecca
Machine:   System: Gigabyte product: N/A
           Mobo: Gigabyte model: Z77-D3H version: x.x Bios: American Megatrends version: F22 date: 11/14/2013
CPU:       Quad core Intel Core i5-3570K CPU (-MCP-) cache: 6144 KB flags: (lm nx sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx) bmips: 27225.2 
           Clock Speeds: 1: 1643.289 MHz 2: 2032.164 MHz 3: 1630.937 MHz 4: 1601.718 MHz
Graphics:  Card: NVIDIA GF114 [GeForce GTX 560 Ti] bus-ID: 01:00.0 X.Org: 1.15.1 driver: nvidia Resolution: 1920x1200@60.0hz 
           GLX Renderer: GeForce GTX 560 Ti/PCIe/SSE2 GLX Version: 4.5.0 NVIDIA 346.59 Direct Rendering: Yes
Audio:     Card-1: NVIDIA GF114 HDMI Audio Controller driver: snd_hda_intel bus-ID: 01:00.1
           Card-2: Intel 7 Series/C210 Series Chipset Family High Definition Audio Controller driver: snd_hda_intel bus-ID: 00:1b.0
           Card-3: Microsoft LifeCam HD-3000 driver: USB Audio usb-ID: 045e:0779
           Sound: Advanced Linux Sound Architecture ver: k3.16.0-37-generic
Network:   Card: Qualcomm Atheros AR8151 v2.0 Gigabit Ethernet driver: atl1c ver: 1.0.1.1-NAPI port: d000 bus-ID: 05:00.0
           IF: eth0 state: up speed: 1000 Mbps duplex: full mac: <filter>
Drives:    HDD Total Size: 1256.3GB (12.7% used) 1: id: /dev/sda model: TOSHIBA_THNSNH25 size: 256.1GB 
           2: id: /dev/sdb model: ST31000528AS size: 1000.2GB 
Partition: ID: / size: 9.1G used: 6.5G (75%) fs: ext4 ID: /home size: 18G used: 3.6G (22%) fs: ext4 
           ID: swap-1 size: 4.56GB used: 0.00GB (0%) fs: swap 
RAID:      No RAID devices detected - /proc/mdstat and md_mod kernel raid module present
Sensors:   System Temperatures: cpu: 29.8C mobo: 27.8C gpu: 0.0:43C 
           Fan Speeds (in rpm): cpu: N/A 
Info:      Processes: 179 Uptime: 7 min Memory: 947.5/7943.6MB Runlevel: 2 Gcc sys: 4.8.2 Client: Shell inxi: 1.8.4 
```

**Comment by emiliedancer** [comment]
Just updated to Cinnamon 2.6.7 and got the SAME error again.  No monitors file this time to delete but disabling xrandr as per pickardjoe's comment on 5 Feb did the trick.  Not sure what the consequences are of disabling that, but nice to have a sane looking desktop again.  I can post the log files again but they are basically the same as my first two posts.  Same trap, same settings error, same whale.  :)

**Comment by sddfdds** [comment]
I am getting this as well with cinnamon 2.6.7 (although I am using the ubuntu mainline kernel 4.0.4, and nvidia 352 from xorg edgers at the moment, I experienced the same issue with Mint's 3.16 and nvidia-331) the xrandr trick does work to restore the desktop to normal, but for me personally, I need cinnamon to change the resolution/multi-monitor stuff, not nvidia-settings, due to the quirkiness of my laptop (Lenovo Z70-80) which has the display output powered by the integrated intel graphics as opposed to nvidia

**Comment by clefebvre** [comment]
Fixed by https://github.com/linuxmint/cinnamon-desktop/commit/dab0d63d766b102d6d54e44e69956d7fe0b9b73f

**Comment by PragTob** [comment]
thank you @clefebvre ! :)

**Comment by clefebvre** [comment]
You're welcome. I eventually got sick of hearing about nvidia-prime issues so I bought an optimus laptop :)

- nvidia 346 was backported also.
- So far prime works but you need to reboot (I think MDM can be improved a bit slightly so you can switch with a simple logout vs a reboot)
- It should be pretty easy to make an applet to switch between the cards, I'll have a look at that too.

**Comment by jedavids** [comment]
Hi Clem,

Thx so much for taking care of this issue (and for all the great work on Cinnamon).  Will this appear in an  update to cinnamon in the near future (I am running cinnamon 2.6.7 currently) or will I need to pull the 2.6.7 source code, apply the patch and create a custom build?

**Comment by Clockwerk300** [comment]
Thank you!

**Comment by maxkl** [comment]
Thank you very much! How long do I have to wait for this to get into a stable release?

**Comment by jedavids** [comment]
Any update on picking this up in a release?  Thanks again!

**Comment by Tectract** [comment]
This bug is apparently upstream from Cinnamon, because it's also occuring in mate-settings-daemon, and the root cause and fix for this problem is still unkown and unpatched, so imho, this issue should remain open until the cause and a patch is at least discovered and reported upstream in the right place.

See here for thread about this same issue in mate-settings-daemon:

https://bugs.launchpad.net/linuxmint/+bug/1471254
